hazelcast/src/main/java/com/hazelcast/spi/Operation.java

```diff
@@ -571,15 +571,20 @@ public final void readData(ObjectDataInput in) throws IOException {
 
     protected abstract void readInternal(ObjectDataInput in) throws IOException;
 
+    protected void toString(StringBuilder sb) {
+    }
+
```

Comment:
Yep, that will work fine. The convention is just to call super.toString(sb) in the implementation of this abstract method in every subclass. If it's not done, everything up the inheritance hierarchy will not be included in the toString(sb) output. 
But I guess it's fine - we should just put it in the javadoc.